DataScience/GpxWavLabeler.ipynb

```diff
@@ -0,0 +1,356 @@
+   "execution_count": 70,
+   "id": "eda2432e-5936-4be3-b5b7-1ffdf5545481",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Identify the interval during which any source emitted\n",
+    "src_min_start_t = min([h[\"start_t\"] for h in collection[\"sources\"]]) * 1000  # [ms]\n",
+    "src_max_stop_t = max([h[\"stop_t\"] for h in collection[\"sources\"]]) * 1000  # [ms]\n",
+    "\n",
+    "# Identify the interval during which all hydrophone collected\n",
+    "hyd_max_start_t = (\n",
+    "    max([h[\"start_t\"] for h in collection[\"hydrophones\"]]) * 1000\n",
+    ")  # [ms]\n",
+    "hyd_min_stop_t = (\n",
+    "    min([h[\"stop_t\"] for h in collection[\"hydrophones\"]]) * 1000\n",
+    ")  # [ms]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 71,
+   "id": "7cba8790-790a-4b86-ba3f-c0e717bc27ba",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2022-04-11 08:35:59,862 - GpxWavLabeler - INFO - Loading data/sampling.json\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Load file describing the sampling cases\n",
+    "# NOTE: Edit this file to change parameters\n",
+    "sampling = gwl.load_json_file(sampling_filepath)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 72,
+   "id": "6fc43cad-f7b3-486b-b1f2-ae86ed2e0aff",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2022-04-11 08:36:00,413 - GpxWavLabeler - INFO - Parsing /Users/raymondleclair/Data/AISonobuoy/suuntoapp-Motorsports-2022-02-22T18-09-01Z-track.gpx\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Consider the first source\n",
+    "source = collection[\"sources\"][0]\n",
+    "gpx_path = Path(data_home) / source[\"name\"]\n",
+    "gpx = gwl.parse_source_gpx_file(gpx_path)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 73,
+   "id": "33082968-1e64-4edd-a50a-b6b9de433ef6",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2022-04-11 08:36:01,002 - GpxWavLabeler - INFO - Getting /Users/raymondleclair/Data/AISonobuoy/Unit-01.WAV\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Consider each hydrophone\n",
+    "hydrophone = collection[\"hydrophones\"][0]\n",
+    "wav_path = Path(data_home) / hydrophone[\"name\"]\n",
+    "audio = gwl.get_hydrophone_wav_file(wav_path)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 74,
+   "id": "1b64a052-0ee5-4407-aec0-c32a1905996c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Export audio with no source present, if it exists\n",
+    "if src_max_stop_t < hyd_min_stop_t:\n",
+    "    gwl.export_audio_interval(\n",
+    "        audio,\n",
+    "        src_max_stop_t,\n",
+    "        hyd_min_stop_t,\n",
+    "        Path(clip_home) / f\"{Path(hydrophone['name'].lower()).stem}-no-source.wav\",\n",
```

Comment:
```no-source``` is a label? Are there other labelled wav files this notebook creates?

Reply:
Yes. As above, and in documentation to be added.